docs/articles_en/openvino_workflow/running_inference_with_openvino/stateful_models_intro.rst

```diff
@@ -0,0 +1,121 @@
+.. {#openvino_docs_OV_UG_stateful_models_intro}
+
+Stateful models and State API
+==============================
+
+.. toctree::
+    :maxdepth: 1
+    :hidden:
+
+    openvino_docs_OV_UG_ways_to_get_stateful_model
+
+What is Stateful Model?
+#######################
+
+Stateful model is a model which stores the specified data between and during model inferences in internal memory space usually called State or Variable.
+In contrast to usual **stateless** model, which purify all the calculated data after each inference call, **stateful**
+model preserve tensors saved in States.
+
+OpenVINO Stateful Model Representation
+######################################
+
+OpenVINO contains ReadValue/Assign operations to make a model Stateful.
+Each pair of ReadValue/Assign operates with State, known also as Variable,
+which is an internal memory buffer to store tensor data during and between model inference calls.
+ReadValue reads data tensor from State and returns it as output, Assign accepts data tensor as input and writes data to State
+to save data for the next inference call.
+
+These entities work as a coherent mechanism and help to achieve the next goals:
+
+1. Speed up execution of Model
+    Data in State is stored in the optimized form for OpenVINO plugins, which helps to execute model effectively.
+    But often requesting data from State might reduce the expected performance gains and even lead to losses,
+    so, it is expected that State mechanism will be used when data stored in State is not accessed frequently.
```

Comment:
We need to be more careful with performance promises. It really depends on the model architecture (patterns used between ReadValue and Assign ops in the graph). Please be more specific and state that performance improvements are expected specifically for large language models produced by Optimum-Intel path and for other models it may vary.

Reply:
I mean even state is not accessed and developer plays by the rules, it may impact performance for some architectures. It is hard to explain when it happens, it is better to point to text generation with model produced by Optimum-Intel where we are definitely expect performance gain.